File: Kiip/Kiip-Android-SDK-v1.0.14-2/kiip-example/src/me/kiip/example/ExampleActivity.java
package me.kiip.example;

import java.util.LinkedList;
import java.util.List;
import java.util.Map;

import me.kiip.api.Kiip;
import me.kiip.api.Kiip.Position;
import me.kiip.api.Kiip.RequestListener;
import me.kiip.api.Kiip.ViewListener;
import me.kiip.api.KiipException;
import me.kiip.api.Resource;
import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.KeyEvent;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.inputmethod.EditorInfo;
import android.widget.Button;
import android.widget.CompoundButton;
import android.widget.CompoundButton.OnCheckedChangeListener;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.TextView.OnEditorActionListener;
import android.widget.Toast;
import android.widget.ToggleButton;

public class ExampleActivity extends Activity
    implements OnClickListener, ViewListener, OnEditorActionListener {
    private static final String TAG = "example";

    private Button mUnlockAchievement, mSaveLeaderboard;

    private EditText mAchievementId,
        mLeaderboardId;

    private ToggleButton mPositionToggle,
        mRewardActionToggle;

    private LinkedList<Resource> mResources = new LinkedList<Resource>();

    /** Called when the activity is first created. */
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setTitle(this.getString(R.string.app_name) + " " + Kiip.VERSION_NAME);
        setContentView(R.layout.main);

        Button button;

        button = (Button)this.findViewById(R.id.getActivePromos);
        button.setOnClickListener(this);

        mAchievementId = (EditText)this.findViewById(R.id.achievementId);
        mAchievementId.setOnEditorActionListener(this);
        mAchievementId.setImeOptions(EditorInfo.IME_ACTION_DONE);
        mLeaderboardId = (EditText)this.findViewById(R.id.leaderboardId);
        mLeaderboardId.setOnEditorActionListener(this);
        mLeaderboardId.setImeOptions(EditorInfo.IME_ACTION_DONE);

        mUnlockAchievement = (Button)this.findViewById(R.id.unlockAchievement);
        mUnlockAchievement.setOnClickListener(this);

        mSaveLeaderboard = (Button)this.findViewById(R.id.saveLeaderboard);
        mSaveLeaderboard.setOnClickListener(this);

        button = (Button)this.findViewById(R.id.showNotification);
        button.setOnClickListener(this);

        button = (Button)this.findViewById(R.id.showFullscreen);
        button.setOnClickListener(this);

        mRewardActionToggle = (ToggleButton)this.findViewById(R.id.rewardActionToggle);
        mRewardActionToggle.setOnCheckedChangeListener(new OnCheckedChangeListener() {
            @Override public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                mPositionToggle.setEnabled(isChecked);
            }
        });
        mPositionToggle = (ToggleButton)this.findViewById(R.id.positionToggle);

        button = (Button)this.findViewById(R.id.newActivity);
        button.setOnClickListener(this);

        if(savedInstanceState != null) {
            // Restore EditText fields
            mAchievementId.setText(savedInstanceState.getString("achievement_id"));
            mLeaderboardId.setText(savedInstanceState.getString("leaderboard_id"));
        }
    }

    @Override
    protected void onStart() {
        super.onStart();

        Kiip.getInstance().startSession(
                this,
                ((ExampleApplication) getApplication()).startSessionListener);
    }

    @Override
    protected void onStop() {
        super.onStop();

        Kiip.getInstance().endSession(
                this,
                ((ExampleApplication) getApplication()).endSessionListener);
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        // Save EditText fields when rotated
        outState.putString("achievement_id", mAchievementId.getText().toString());
        outState.putString("leaderboard_id", mLeaderboardId.getText().toString());
    }

    @Override
    public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
        switch (v.getId()) {
            case R.id.achievementId:
                mUnlockAchievement.performClick();
                break;
            case R.id.leaderboardId:
                mSaveLeaderboard.performClick();
                break;
            default:
                return false;
        }
        return false;
    }

    @Override
    public void onClick(View v) {
        Kiip manager = Kiip.getInstance();

        switch (v.getId()) {
            case R.id.getActivePromos:
                manager.getActivePromos(mActivePromosListener);
                break;

            case R.id.unlockAchievement:
                manager.unlockAchievement(mAchievementId.getText().toString(), mRequestListener);
                break;

            case R.id.saveLeaderboard:
                manager.saveLeaderboard(mLeaderboardId.getText().toString(), 100, mRequestListener);
                break;

            case R.id.showNotification:
                if (mResources.size() > 0) {
                    Resource resource;
                    while(mResources.size() > 0) {
                        resource = mResources.removeFirst();
                        manager.showResource(resource);
                    }
                }
                break;

            case R.id.showFullscreen:
                if (mResources.size() > 0) {
                    Resource resource;
                    toast("Showing Fullscreen (" + mResources.size() + ")");
                    while(mResources.size() > 0) {
                        resource = mResources.removeFirst();
                        resource.position = Position.FULLSCREEN;
                        manager.showResource(resource);
                    }
                }
                break;

            case R.id.newActivity:
                Intent intent = new Intent(this, ExampleActivity.class);
                startActivity(intent);
                break;
        }
    }

    private RequestListener<List<Map<String, String>>> mActivePromosListener = new RequestListener<List<Map<String, String>>>() {
        @Override
        public void onFinished(Kiip manager, List<Map<String, String>> response) {
            toast(response.toString());
        }

        @Override
        public void onError(Kiip manager, KiipException error) {
            toast("error (" + error.getCode() + ") " + error.getMessage());
        }
    };

    private RequestListener<Resource> mRequestListener = new RequestListener<Resource>() {
        @Override
        public void onFinished(Kiip manager, Resource response) {
            if (response != null) {
                if (mRewardActionToggle.isChecked()) {
                    manager.showResource(response);
                } else {
                    toast("Reward Queued");
                    mResources.add(response);
                }
            } else {
               toast("No Reward");
            }
        }

        @Override
        public void onError(Kiip manager, KiipException error) {
            toast("error (" + error.getCode() + ") " + error.getMessage());
        }
    };

//==============================================================================
// KPViewListener callbacks
//==============================================================================

    @Override
    public void onNotificationDidShow(Resource resource) {
        toast("NotificationDidShow");
    }

    @Override
    public void onNotificationDidDismiss(Resource resource, boolean clicked) {
        toast("NotificationDidDismiss(" + clicked + ")");
    }

    @Override
    public void onFullscreenDidShow(Resource resource) {
        toast("FullscreenDidShow");
    }

    @Override
    public void onFullscreenDidDismiss(Resource resource) {
        toast("FullscreenDidDismiss");
    }

//==============================================================================
// Util methods
//==============================================================================

    private void toast(String message) {
        Log.v(TAG, message);
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
    }
}

File: Kiip/KiipSampleApplication/ExampleActivity.cs
using System;

using Android.App;
using Android.Content;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Android.OS;
using System.Collections.Generic;
using Android.Views.InputMethods;
using ME.Kiip.Api;
using Android.Util;
using Android.Telephony;

namespace KiipSampleApplication
{
    [Activity(Label = "KiipSampleApplication", MainLauncher = true, Icon = "@drawable/icon")]
    public class ExampleActivity : Activity, Android.Widget.TextView.IOnEditorActionListener, Android.Views.View.IOnClickListener, ME.Kiip.Api.Kiip.IViewListener
    {

        private static readonly string TAG = "example";

        private Button mUnlockAchievement, mSaveLeaderboard, mShowNotification, mShowFullscreen, mGetActivePromos, mNewActivity;

        private EditText mAchievementId, mLeaderboardId;

        private static ToggleButton mPositionToggle, mRewardActionToggle;

        private List<ME.Kiip.Api.Resource> mResources = new List<ME.Kiip.Api.Resource>();

        private RewardRequestListener mRewardsListener;
        private ActivePromosRequestListener mActivePromosListener;

        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            // Set our view from the "main" layout resource
            SetContentView(Resource.Layout.Main);

            SetTitle(Resource.String.app_name);

            mGetActivePromos = (Button)FindViewById(Resource.Id.getActivePromos);
            mGetActivePromos.SetOnClickListener(this);

            mAchievementId = (EditText)FindViewById(Resource.Id.achievementId);
            mAchievementId.SetOnEditorActionListener(this);
            mAchievementId.ImeOptions = ImeAction.Done;
            mLeaderboardId = (EditText)FindViewById(Resource.Id.leaderboardId);
            mLeaderboardId.SetOnEditorActionListener(this);
            mLeaderboardId.ImeOptions = ImeAction.Done;

            mUnlockAchievement = (Button)FindViewById(Resource.Id.unlockAchievement);
            mUnlockAchievement.SetOnClickListener(this);

            mSaveLeaderboard = (Button)FindViewById(Resource.Id.saveLeaderboard);
            mSaveLeaderboard.SetOnClickListener(this);

            mShowNotification = (Button)FindViewById(Resource.Id.showNotification);
            mShowNotification.SetOnClickListener(this);

            mShowFullscreen = (Button)FindViewById(Resource.Id.showFullscreen);
            mShowFullscreen.SetOnClickListener(this);

            mRewardActionToggle = (ToggleButton)FindViewById(Resource.Id.rewardActionToggle);
            mPositionToggle = (ToggleButton)FindViewById(Resource.Id.positionToggle);

            mNewActivity = (Button)FindViewById(Resource.Id.newActivity);
            mNewActivity.SetOnClickListener(this);

            if (bundle != null)
            {
                // Restore EditText fields
                mAchievementId.Text = bundle.GetString("achievement_id");
                mLeaderboardId.Text = bundle.GetString("leaderboard_id");
            }
            else
            {
                mAchievementId.Text = "com.jmawebtechnologies.gemfindermt.beatgame";
                mLeaderboardId.Text = "com.jmawebtechnologies.gemfindermt.leaderboard.onethousand";
            }

            mActivePromosListener = new ActivePromosRequestListener(this);
            mRewardsListener = new RewardRequestListener(this);
        }


        public bool OnEditorAction(TextView v, ImeAction actionId, KeyEvent e)
        {
            switch (v.Id)
            {
                case Resource.Id.achievementId:
                    mUnlockAchievement.PerformClick();
                    break;
                case Resource.Id.leaderboardId:
                    mSaveLeaderboard.PerformClick();
                    break;
                default:
                    return false;
            }
            return false;
        }

        protected override void OnStart()
        {
            base.OnStart();
            Kiip.Instance.StartSession(
                this, ((ExampleApplication)Application).startSessionListener);
        }

        protected override void OnStop()
        {
            base.OnStop();
            Kiip.Instance.StartSession(
                this, ((ExampleApplication)Application).endSessionListener);

        }

        protected override void OnSaveInstanceState(Bundle bundle)
        {
            base.OnSaveInstanceState(bundle);
            // Save EditText fields when rotated
            bundle.PutString("achievement_id", mAchievementId.Text);
            bundle.PutString("leaderboard_id", mLeaderboardId.Text);
        }

        class OnCheckedChangedListener : Android.Widget.RadioGroup.IOnCheckedChangeListener
        {

            public void OnCheckedChanged(RadioGroup group, int checkedId)
            {
                if (checkedId == 0)
                {
                    mPositionToggle.Enabled = false;
                }
                else
                {
                    mPositionToggle.Enabled = true;
                }
            }

            public void Dispose()
            {

            }

            public IntPtr Handle { get; set; }
        }

        class RewardRequestListener : Java.Lang.Object, Kiip.IRequestListener
        {
            ExampleActivity example = new ExampleActivity();

            public RewardRequestListener(ExampleActivity example)
            {
                this.example = example;
            }

            public void OnError(Kiip p0, KiipException p1)
            {
                example.toast("error (" + p1.Code + ") " + p1.Message);
            }

            public void OnFinished(Kiip p0, Java.Lang.Object p1)
            {
                ME.Kiip.Api.Resource response = p1.JavaCast<ME.Kiip.Api.Resource>();
                if (response != null)
                {
                    if (mRewardActionToggle.Checked)
                    {
                        p0.ShowResource(response);
                    }
                    else
                    {
                        example.toast("Reward Queued");
                        example.mResources.Add(response);
                    }
                }
                else
                {
                    example.toast("No Reward");
                }
            }
        }

        class ActivePromosRequestListener : Java.Lang.Object, Kiip.IRequestListener
        {
            ExampleActivity example = new ExampleActivity();

            public ActivePromosRequestListener(ExampleActivity example)
            {
                this.example = example;
            }

            public void OnError(Kiip p0, KiipException p1)
            {
                example.toast("error (" + p1.Code + ") " + p1.Message);
      
            }

            public void OnFinished(Kiip p0, Java.Lang.Object p1)
            {
                ME.Kiip.Api.Resource response = p1.JavaCast<ME.Kiip.Api.Resource>();
                example.toast(response.ToString());
            }
        }

        public void OnClick(View v)
        {
            Kiip manager = Kiip.Instance;

            switch (v.Id)
            {
                case Resource.Id.getActivePromos:
                    manager.GetActivePromos(mActivePromosListener);
                    break;

                case Resource.Id.unlockAchievement:
                    manager.UnlockAchievement(mAchievementId.Text, new RewardRequestListener(this){});
                    break;

                case Resource.Id.saveLeaderboard:
                    manager.SaveLeaderboard(mLeaderboardId.Text, 100, mRewardsListener);
                    break;

                case Resource.Id.showNotification:
                    if (mResources.Count > 0)
                    {
                        ME.Kiip.Api.Resource resource = mResources[0];
                        manager.ShowResource(resource);
                    }
                    break;

                case Resource.Id.showFullscreen:
                    if (mResources.Count > 0)
                    {
                        toast("Showing Fullscreen (" + mResources.Count + ")");

                        ME.Kiip.Api.Resource resource = mResources[0];
                        resource.Position = ME.Kiip.Api.Kiip.Position.Fullscreen;
                        manager.ShowResource(resource);
                    }
                    break;

                case Resource.Id.newActivity:
                    Intent intent = new Intent(this, this.Class);
                    StartActivity(intent);
                    break;
            }
        }
    
        //==============================================================================
        // Util methods
        //==============================================================================

        private void toast(string message)
        {
            Log.Verbose(TAG, message);
            Toast.MakeText(this, message, ToastLength.Short).Show();
        }

        //==============================================================================
        // KPViewListener callbacks
        //==============================================================================

        public void OnFullscreenDidDismiss(ME.Kiip.Api.Resource p0)
        {
            toast("FullscreenDidDismiss");
        }

        public void OnFullscreenDidShow(ME.Kiip.Api.Resource p0)
        {
            toast("FullscreenDidShow");
        }

        public void OnNotificationDidDismiss(ME.Kiip.Api.Resource p0, bool p1)
        {
            toast("NotificationDidDismiss(" + p1 + ")");
        }

        public void OnNotificationDidShow(ME.Kiip.Api.Resource p0)
        {
            toast("Notification did show");
        }
    }

}

